Scripts/RemediationScripts/Get-ManagementCertificates.ps1

```diff
@@ -0,0 +1,352 @@
+        $PerformPreReqCheck
+        )
+
+
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 1 of 2] Prepare to get management certificates in Subscription: [$($SubscriptionId)]"
+    Write-Host $([Constants]::SingleDashLine)
+
+    if ($PerformPreReqCheck)
+    {
+        try
+        {
+            Write-Host "Setting up prerequisites..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            Setup-Prerequisites
+            Write-Host "Completed setting up prerequisites." -ForegroundColor $([Constants]::MessageType.Update)
+            Write-Host $([Constants]::SingleDashLine)
+        }
+        catch
+        {
+            Write-Host "Error occurred while setting up prerequisites. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+            return
+        }
+    }
+    
+    
+     $context = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop
+
+     Write-Host "Subscription Name: [$($context.Subscription.Name)]"
+     Write-Host "Subscription ID: [$($context.Subscription.SubscriptionId)]"
+     Write-Host "Account Name: [$($context.Account.Id)]"
+     Write-Host "Account Type: [$($context.Account.Type)]"
+     Write-Host $([Constants]::SingleDashLine)
+    
+    # To get the Resource URL from environment
+    $ResourceAppIdURI = GetServiceManagementUrl
+
+    # Using Get-AzAccessToken to get access token for fetching management certificates.
+    $ClassicAccessToken= (Get-AzAccessToken -ResourceUrl $ResourceAppIdURI).Token
+    
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 2 of 2] Fetching  management certificates in Subscription: [$($SubscriptionId)]"
+    Write-Host $([Constants]::SingleDashLine)
+
+    if($null -ne $ClassicAccessToken)
+    {
+        $header = "Bearer " + $ClassicAccessToken
+        $headers = @{"Authorization"=$header;"Content-Type"="application/json"; "x-ms-version" ="2013-08-01"}
+
+        $uri = [string]::Format("{0}/{1}/certificates",$ResourceAppIdURI,$SubscriptionId)
+        $mgmtCertsResponse = Invoke-WebRequest -Method GET -Uri $uri -Headers $headers -UseBasicParsing
+        if($mgmtCertsResponse.StatusCode -ge 200 -and $mgmtCertsResponse.StatusCode -le 399)
+        {
+            if($null -ne $mgmtCertsResponse.Content)
+            {
+                [xml] $mgmtCerts = $mgmtCertsResponse.Content;
+                if($null -ne $mgmtCerts -and [Helpers]::CheckMember($mgmtCerts, "SubscriptionCertificates.SubscriptionCertificate"))
+                {
+                    # Includes all the management certificates from the subscription
+                    $ManagementCertificates = [ManagementCertificate]::ListManagementCertificates($mgmtCerts.SubscriptionCertificates.SubscriptionCertificate)
+                
+                    $context = Get-AzContext
```

Comment:
Why context check is needed here

Reply:
Removed AzContext check as it is already done above.